docs/design/architecture.md

```diff
@@ -0,0 +1,29 @@
+# Architecture 
+Kubernetes Console project consists of two main components. They are called here the
+frontend and the backend.
+
+The frontend is a single page web application that runs in a browser. It fetches all its
+business data from the backend using standard HTTP methods. It does not implement business logic;
+it only presents fetched data and sends requests to the backend for actions.
+
+The backend runs in a Kubernetes cluster as a kubernetes service. Alternatively, it may run anywhere
+outside of the cluster, given that it can connect to the master. The backend is a HTTP server that
+proxies data requests to appropriate remote backends (e.g., k8s apiserver or heapster) or implements
```

Comment:
Authorization is a problem for the dashboard, monitoring (heapster, kubedash), and logging (elasticsearch). It would be useful for someone to think about how it should work.

One idea a couple of us have discussed for the dashboard is to launch dashboards running on behalf of the user on demand, rather than running the dashboard as a shared service. @erictune